enhancements/sig-policy/72-content-to-ansilbe-job/README.md

```diff
@@ -0,0 +1,174 @@
+launches a predefined Ansible playbook against an inventory of hosts. During this creation, we can pass the 
+additional information to the Ansible job as `extra_vars`.
+
+The current limitation is that we only pass the non-compliant cluster as `extra_vars`, which isn't enough to 
+describe the policy violation status. The following fields could be added:
+1. The violated policy name
+2. The policy violation message
+3. The replicated policy status for each non-compliant cluster
+
+
+### Goals
+
+Adding the name of the violated policy, the violation message, and the replicated policy status for each 
+non-compliant cluster into the Ansible Job.
+
+### Non-Goals
+
+- None at this time
+
+## Proposal
+Policy automation runs on the hub cluster, it should not get the propagated policy status from the managed cluster,
+which requires Search or Observability dependency. The same design also applies to the hub of the cluster scenario.
+
+### New Content Structure
+
+Current `extra_vars` has a field `target_clusters` contains non-compliant cluster name list:
+```json
+{
+  "target_clusters": [
+    "cluster1", 
+    "cluster2", 
+    "cluster3",
+  ]
+}
+```
+
+The new content structure will pass more information:
+```json
+{
+  "target_clusters": [
+    "cluster1", 
+    "cluster2", 
+    "cluster3",
+  ],
+  "policy_name": "policy1",
+  "violation_message": "NonCompliant; violation -  xxxxxx",
+  "policy_violation_context" : {
+    "local-cluster": {
+      "compliant": "NonCompliant",
+      "details": [
+        {
+          "compliant": "NonCompliant",
+          "history": [
+            {
+              "lastTimestamp": "2022-09-08T16:03:23Z",
+              "message": "NonCompliant; violation - pods not found: [testpod] in namespace default missing"
+            },
+          ],
+        },
+      ]
+    },
+    "cluster2": {...},
+    },
+}
+```
+
+- `policy_name` is the name of the non-compliance policy that triggers the ansible job.
+  `violation_message` is the latest policy violation message. 
+  They are from the root policy on the hub cluster. 
+  The field path of `policy_name` is `policy.metadata.name`.
+  The field path of `violation_message` is `policy.policyStatus.details[0].history[0].message`
+
+- `policy_violation_context` is a map. The keys are non-compliant cluster names, and the value is the raw replicated 
+  policy status for each non-compliant cluster.
+  The policy automation looks at the `cluster namespaces` of the hub cluster and gets the replicated policies. Then it filters out the non-compliant replicated policies and puts the status into the `policy_violation_context` map.
+
+  Suppose the policy automation runs on the managed cluster. In that case, the policy automation looks at the propagated policy on the managed cluster, then `policy_violation_context` would be a map that only contains the propagated policy status.
```

Comment:
Just to clarify since the PolicyAutomation runs on the hub.
```suggestion
  Suppose the policy automation runs for the managed cluster. In that case, the policy automation looks at the propagated policy of the managed cluster, then `policy_violation_context` would be a map that only contains the propagated policy status.
```